### DISCUSSION TITLE
[FEATURE] codereview功能如何实现？

### DISCUSSION THREAD

**Discussion by muwoo** [opensumi/core#1783] [General]
看文档上说可以实现codereview能力。但是有么有具体的demo可以简单介绍一下使用方式？
![image](https://user-images.githubusercontent.com/21073039/156687387-d5724519-b1f2-43b5-8358-fcbd9edbcba2.png)

**Comment by hacke2** [comment]
CR  主要分三块：
1. 使用 Tree API 实现变更树
2. 如果是纯前端版注册一个处理 git 协议的 editorDocumentModelContentProvider 用了定义 git 协议如何打开，带容器的不需要，git 插件会注册
3. 使用 comments api 给对应变更文件添加评论，也可以用模块的 api 做更多的拓展（例如给编辑器 decoration 上加当前评论者的头像、替换整个内置评论组件等）

上面说的内容大部分逻辑写在了上层集成放里，目前还没下沉到 opensumi，我写个文档和最佳实践示例

**Reply by relife957** [reply]
大佬这个文档现在有了吗

**Reply by hacke2** [reply]
再等等，之后会开源一个基于 OpenSumi 的 Lite 解决方案评审组件，上面说的这些 API 都封装在了组件里，你们无需关心。
可以在这里体验下 https://www.gitlink.org.cn/hacke2/core/pulls/3/review ，预计明年 Q1 开源

**Reply by venciallee** [reply]
请问 Lite IDE CR 组件和 Cloud 的CR 组件有啥区别呢？

**Reply by hacke2** [reply]
底层都是使用 OpenSumi，不过 Cloud IDE CR 组件正对与代码服务场景有更多的封装

**Comment by muwoo** [comment]
@hacke2 感谢回复！非常期待这个功能的最佳实践示例。之前我也在蚂蚁用了你们的这个codereview，感觉很棒！现在我们也希望可以集成这个能力，目前来看文档稍微欠缺了点。持续关注中！感谢开源！

**Comment by haozl1129** [comment]
请问，现在有示例了吗？
